sources/newRif/rif_gemara_matcher.py

```diff
@@ -0,0 +1,52 @@
+import re
+import os
+import csv
+import django
+django.setup()
+from functools import partial
+from sefaria.model import *
+from rif_utils import path, remove_metadata, tags_map, get_hebrew_masechet
+from data_utilities.dibur_hamatchil_matcher import match_ref
+from sefaria.system.exceptions import InputError
+
+def base_tokenizer(string):
+    return re.sub(r'<.*?>', '', string).split()
+
+for masechet in list(tags_map):
+    heb_masechet = get_hebrew_masechet(masechet)
+    data = []
+    report = []
+    for root, dirs, files in os.walk(os.getcwd()+'/rif_csv'):
+        for file in files:
+            if masechet in file:
+                with open(path+'/rif_csv/'+file, encoding='utf-8', newline='') as file:
+                    data = list(csv.DictReader(file))
+
+    splitted_data = {}
+    for row in data:
+        if tags_map[masechet]['gemara_refs'] in row['content']:
+            if remove_metadata(row['content'].split(tags_map[masechet]['gemara_refs'])[0], masechet) != '':
+                splitted_data[current_ref].append(row)
+            for ref in re.findall(r'{}(.*?){}'.format(tags_map[masechet]['gemara_refs'], tags_map[masechet]['end_tag']), row['content']):
+                if ref not in list(splitted_data):

```

Comment:
Checking if a key exists in a dictionary is *more* performant (O(1)) than checking if it exists in a list (O(n)). That's in addition to the overhead of creating the list in the first place.